In [1]:
import pandas as pd
from sklearn import linear_model
from sklearn.metrics import r2_score

In [2]:
df = pd.read_csv('housing.csv')

Elimina los máximos

In [3]:
df = df[df.median_house_value != df.median_house_value.max()]

## Crea un modelo de regresión lineal simple usando el ingreso y calcula su R2

In [11]:
model_income = linear_model.LinearRegression()
X_income = df[['median_income']]
target = 'median_house_value'
y = df[target]

model_income.fit(X_income, y)
predictions_income = model_income.predict(X)

r2_score(y_pred=predictions_income, y_true=y)

0.41294861985820197

## Crea un modelo de regresión lineal simple usando la edad de las casas

In [12]:
model_age = linear_model.LinearRegression()
X_age = df[['housing_median_age']]

model_age.fit(X_age, y)
predictions_age = model_age.predict(X_age)

r2_score(y_pred=predictions_age, y_true=y)

0.004616870761556036

## Crea un modelo de regresión lineal multiple usando las variables anteriores

In [13]:
X_multi = df[['median_income', 'housing_median_age']]
model_multiple = linear_model.LinearRegression()

model_multiple.fit(X_multi, y)
predictions_multiple = model_multiple.predict(X_multi)

r2_score(y_pred=predictions_multiple, y_true=y)

0.45171387837376886

Escribe la ecuación del modelo

In [14]:
# Interepto
model_multiple.intercept_

-7997.527316095948

In [15]:
# Coeficientes
model_multiple.coef_

array([42420.02474575,  1567.97684924])

$Precio = -7997.52 + 1567 * edad + 42420 * ingreso$

$$Precio = \beta_0 + \beta_1 * x_1 + \beta_2* x_2$$

$$Precio = -10189 + 1744 * edad + 43169* ingreso$$

### Agrega la variable de households

In [18]:
X_multi2 = df[['median_income', 'housing_median_age', 'households']]
model_multiple2 = linear_model.LinearRegression()

model_multiple2.fit(X_multi2, y)
predictions_multiple2 = model_multiple2.predict(X_multi2)

r2_score(y_pred=predictions_multiple2, y_true=y)

0.4691647656140183

Escribe la ecuación del modelo

In [19]:
model_multiple2.intercept_

-35840.7735441851

In [20]:
model_multiple2.coef_

array([4.25476800e+04, 1.90691959e+03, 3.54211223e+01])

### Agrega todas

In [8]:
### Verific los NA's


In [9]:
# Quita NA's 


In [10]:
# Convertir a one hot encoding 


### Visualizando el modelo, calibracion

### Compara la calibración de todos los modelos anteriores